go/interpreter/lfvm/instruction_logger_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func TestLogger_ExecutesCodeAndLogs(t *testing.T) {
+
+	tests := map[string]struct {
+		code []Instruction
+		want string
+	}{
+		"empty": {},
+		"stop": {
+			code: []Instruction{{STOP, 0}},
+			want: "STOP, 10, -empty-\n",
+		},
+		"multiple codes": {
+			code: []Instruction{{PUSH4, 0}, {DATA, 1}, {STOP, 0}},
+			want: "PUSH4, 10, -empty-\nSTOP, 7, 1\n",
+		},
```

Comment:
can we add a test for an error? 
this would add coverage for line 50 in instruction_logger.go and we wont have to add it later on